src/CIQS-CLI-Extension/azext_ciqs/custom.py

```diff
@@ -30,15 +54,42 @@ def deployDeployment(cmd, deploymentId, subscription=None):
     print('To be implemented')
 
 def viewDeployment(cmd, deploymentId, subscription=None):
-    print('To be implemented')
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = '/api/deployments/' + subscription + '/' + deploymentId
+    return _makeAPICall(cmd, 'GET', path, auth_token=auth_token)
 
 def deleteDeployment(cmd, deploymentId, subscription=None):
-    print('To be implemented')
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    if subscription is None:
+        subscription = get_subscription_id(cmd.cli_ctx)
+    profile = Profile(cli_ctx=cmd.cli_ctx)
+    auth_token = profile.get_raw_token(subscription=subscription)
+    path = '/api/deployments/' + subscription + '/' + deploymentId
+    return _makeAPICall(cmd, 'DELETE', path, auth_token=auth_token)
 
-# group ciqs gallery
+#---------------------------------------------------------------------------------------------
+# sub-group ciqs gallery
+#---------------------------------------------------------------------------------------------
 
 def listTemplates(cmd):
-    print('To be implemented')
+    path = '/api/gallery'
+    return _makeAPICall(cmd, 'GET', path)
```

Comment:
I'll look into that. The expected output by the Azure CLI is JSON, you can add an argument to instead output as a table. I was planning on using their framework to have a custom table in a more readable manner. I am hesitant to force it to output in that manner when the user is expecting output to be JSON.

Reply:
Here is the expectation for all commands in the cli:
https://docs.microsoft.com/en-us/cli/azure/format-output-azure-cli?view=azure-cli-latest

Reply:
I see.. For ListTemplates and ListDeployment I would suggest to display in table format by default. For the other commands, json is fine, such as:

```
[
  {
    "SolutionId": "customer-360",
    "SolutionName": "Customer 360",
    "Owners": ["xyz@microsoft.com", "abc@microsoft.com"],
    "Ingredients": ["SQL Server", "Data Factory"],
    "TryItNow": "https://xyz.abc.com",
    "Deployment link: https://xyz.def.com"
  },
  {
    "SolutionId": "hospital",
    "SolutionName": "Predictive Length of Stay in Hospital",
    "Owners": ["efg@microsoft.com", "rst@microsoft.com"],
    ”Ingredients“: ["SQL Server", "Data Factory"],
    ”TryItNow“: ”https://uvw.abc.com",
    ”Deployment“: https://xyz.lmn.com",
  }
]
```

I expect that it **won't** show all metadata of a template, because it will be too much in a commandline interface; it would be awesome if we can display just enough amount of information and deserialize it as json/table.

Reply:
The default is selected by the user in the Azure CLI. For that reason, I don't believe I can change the default for the command.

Reply:
I can however, change the data that shows up in the table when a table is selected. I've been looking into that. By default it takes the JSON and infers a table, but they have a way to specify that a command has a custom table, where you specify which attributes from the JSON to include or not include.